lib/discovery_dispatcher/indexing_job.rb

```diff
@@ -10,42 +10,47 @@ def perform
       target_url  = get_target_url target, druid
       method      = get_method type, druid
 
-      request_command = build_request_command(druid, method, target_url, target)
-      run_request_command(druid, type, request_command)
+      url = build_request_url(druid, target_url, target)
+      run_request(druid, type, url)
 
       # Request went successfully
       Rails.logger.debug "Completing #{druid_id} for target #{target}"
     end
 
     # @param druid [String] represents the druid on the form of ab123cd4567
-    # @param method [String] it may be put or delete
     # @param target_url [String] the url for the indexing service
     # @param solr_target [Array] solr target for solr core
     # @return [String] RestClient request command
-    def build_request_command(druid, method, target_url, solr_target)
+    def build_request_url(druid, target_url, solr_target)
       solr_target = Array(solr_target)
-      request_command = "RestClient.#{method} \"#{target_url}/items/#{druid}?#{solr_target.to_query('solr_target')}\""
-      request_command = "#{request_command}, \"\"" if method == 'put'
-      request_command
+      "#{target_url}/items/#{druid}?#{solr_target.to_query('solr_target')}"
     end
 
     # It runs the request command
     # @param druid [String] represents the druid on the form of ab123cd4567
     # @param type [String] index or delete
     # @param request_command [String] RestClient request command
-    def run_request_command(druid, type, request_command)
+    def run_request(druid, type, request_url)
       response = nil
       begin
-        response = eval(request_command)
+        conn = Faraday.new(url: request_url)
+        if type == 'index'
+          response = conn.put do |request|
+            request.body = ''
+          end
+        else
+          response = conn.delete
```

Comment:
This is the API call for the sw-indexer.  If the call is for indexing (in the case of new druid or updated druid), the API call is a PUT with the appropriate url which is built in build_request_url. The PUT requires a payload called body which is empty in our scenario. If the call is for a deleted druid, the API call is a DELETE with the appropriate URL from build_request_url. Does that make sense?
@mejackreed